File: 94.binary-tree-inorder-traversal.js
function TreeNode(val, left, right) {
  this.val = (val===undefined ? 0 : val)
  this.left = (left===undefined ? null : left)
  this.right = (right===undefined ? null : right)
}

var inorderTraversal = function(root) {
    const result = [];

    function traverse(node) {
      if (!node) {
        return;
      }
      traverse(node.left);
      result.push(node.val);
      traverse(node.right);
    }
    traverse(root);
    return result;
};

const tree = new TreeNode(1, null, new TreeNode(2, new TreeNode(3), null));
console.log(inorderTraversal(tree)); // Виведе: [1, 3, 2]

File: 94.binary-tree-inorder-traversal.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inorderTraversal(self, root):
        result = []

        def traverse(node):
            if not node:
                return
            traverse(node.left)  # 1. Обходимо ліву гілку
            result.append(node.val)  # 2. Додаємо значення вузла
            traverse(node.right)  # 3. Обходимо праву гілку

        traverse(root)
        return result

tree = TreeNode(val=1, left=None, right=TreeNode(2, TreeNode(3), None))

sol = Solution()
print(sol.inorderTraversal(tree))